data_store/cli.py

```diff
@@ -18,148 +23,292 @@
+            reader = csv.DictReader(csvfile)
+            for row in reader:
+                email = row["email"]
+                full_name = row["full_name"]
+                azure_ad_subject_id = row["azure_ad_subject_id"]
+
+                # Try to get account
+                try:
+                    emailinfo = validate_email(email, check_deliverability=False)
+                    email = emailinfo.normalized
+                except EmailNotValidError as error:
+                    error_message = f"Email<{email}> - Error: {str(error)}"
+                    result["errors"].append(error_message)
+                    click.secho(error_message, fg="red")
+
+                    # Go to next email
+                    continue
+
+                # Try to get account by email
+                try:
+                    response = requests.get(
+                        f"{Config.ACCOUNT_STORE_API_HOST}/accounts",
+                        headers={"Content-Type": "application/json"},
+                        params={"email_address": email},
+                    )
+
+                    response.raise_for_status()
+
+                    account = response.json()
+                except RequestException as error:
+                    if error.response.status_code != 404:
+                        error_message = f"Account<{email}> - Error: {error.response.text}".strip("\n")
+                        result["errors"].append(error_message)
+                        click.secho(error_message, fg="yellow")
+
+                        # Go to next email
+                        continue
+
+                    # If account not found, set account to None and we create it
+                    account = None
+
+                if account is None:
+                    # Create account
+                    try:
+                        response = requests.post(
+                            f"{Config.ACCOUNT_STORE_API_HOST}/accounts",
+                            headers={"Content-Type": "application/json"},
+                            json={"email_address": email, "azure_ad_subject_id": azure_ad_subject_id},
+                        )
+
+                        response.raise_for_status()
+
+                        account = response.json()
+                        result["created"] += 1
+                        click.secho(f"Account<{email}> - Account created", fg="yellow")
+                    except RequestException as error:
+                        error_message = f"Account<{email}> - Error: {error.response.text}".strip("\n")
+                        result["errors"].append(error_message)
+                        click.secho(error_message, fg="red")
+
+                        # Go to next email
+                        continue
+
+                # Update account with roles
+                data = {
+                    "email_address": email,
+                    "azure_ad_subject_id": account.get("azure_ad_subject_id"),
```

Comment:
If the `account` retrieved doesn't have `azure_ad_subject_id` - then this call fails? If `azure_ad_subject_id` is required in the CSV, then that should probably be used as a fallback value here for if the account already exists?

It is possible for accounts to exist without `azure_ad_subject_id` being set (whether that's a valid state in our system, I'm less sure - but I believe so).

Reply:
I think I just find this whole thing around `azure_ad_subject_id` a bit weird.

If it's not already on the account in account-store, how are we going to have it to put in our CSV? And if we don't have it in either place, then the PUT just won't go through? I feel like we either need to be able to create accounts with roles in one go, or the update account PUT endpoint should not require `azure_ad_subject_id`.

I know you and @gidsg spoke about this already - what am I missing? It feels like I must be missing something.

Reply:
I changed this back. Now that `azure_ad_subject_id` defaults to random uuid, falling back on that value will result in a broken request. 

I think there are work arounds, but until we use this for non-test purposes, it won't affect us.